### ISSUE TITLE
Cannot initialize idp with IdentityProviderManager

### ISSUE THREAD

**Issue by Biboba** [Esri/arcgis-python-api#827]
**Describe the bug**
When trying to configure ArcGIS Enterprise Idp, the following error occurs:
```
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
**To Reproduce**
```python
from arcgis.gis import GIS

# Initialize GIS: no idp should be configured
portal_url = "https://arcgisenterprise.company.com/webcontext"
portal_user = "siteadmin"
portal_pwd = "mypassword"

ent_gis = GIS(portal_url, portal_user, portal_pwd, verify_cert=False)


# Display idp returns an error (cf. below)
ent_gis.admin.idp.properties

# Trying to configure it, returns the same error:
ent_gis.admin.idp.configuration = {
   'name': 'IDP Account',
   'idpMetadataFile' : 'C:/Temp/federationmetadata.xml',
   'entityId': "arcgisenterprise.company.com.webcontext",
   'signUpMode': 'Automatic',
   'encryptionSupported': False,
   'supportSignedRequest': False,
   'useSHA256': False,
   'supportsLogoutRequest': False,
   'updateProfileAtSignin': True,
   'updateGroupsAtSignin': True
}

# Documentation
# https://developers.arcgis.com/python/api-reference/arcgis.gis.admin.html?#identityprovidermanager

```
error:
```python
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
<ipython-input-144-595781e15061> in <module>
----> 1 ent_gis.admin.idp.properties

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\arcgis\gis\admin\_idp.py in properties(self)
     40             params = {'f':'json'}
     41             res = self._gis._con.get(path=self._url,
---> 42                                      params=params)
     43             try:
     44                 self._properties = PropertyMap(res)

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\arcgis\gis\_impl\_con\_connection.py in get(self, path, params, **kwargs)
    423                 'An unknown error occurred: %s' % traceback.format_exc())
    424 
--> 425         return self._handle_response(resp, file_name, out_path, try_json, force_bytes=kwargs.pop('force_bytes', False))
    426     #----------------------------------------------------------------------
    427     def _handle_response(self, resp, file_name, out_path, try_json, force_bytes=False):

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\arcgis\gis\_impl\_con\_connection.py in _handle_response(self, resp, file_name, out_path, try_json, force_bytes)
    495                     raise Exception(resp['error'])
    496             else:
--> 497                 data = resp.json()
    498             #if 'error' in data:
    499                 #raise Exception(data['error'])

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\site-packages\requests\models.py in json(self, **kwargs)
    896                     # used.
    897                     pass
--> 898         return complexjson.loads(self.text, **kwargs)
    899 
    900     @property

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\json\__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    352             parse_int is None and parse_float is None and
    353             parse_constant is None and object_pairs_hook is None and not kw):
--> 354         return _default_decoder.decode(s)
    355     if cls is None:
    356         cls = JSONDecoder

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\json\decoder.py in decode(self, s, _w)
    337 
    338         """
--> 339         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    340         end = _w(s, end).end()
    341         if end != len(s):

C:\Program Files\ArcGIS\Pro\bin\Python\envs\arcgispro-py3\lib\json\decoder.py in raw_decode(self, s, idx)
    355             obj, end = self.scan_once(s, idx)
    356         except StopIteration as err:
--> 357             raise JSONDecodeError("Expecting value", s, err.value) from None
    358         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
**Screenshots**

**Expected behavior**
Idp properties should be displayed and configurable

**Platform (please complete the following information):**
 - OS: [e.g. iOS] Window
 - Browser [e.g. chrome, safari]: chrome
 - Python API Version : 1.8.1

**Additional context**

I think it is due to the fact that:
https://arcgisenterprise.company.com/webcontext/sharing/portals/self/idp?f=json returns 200 but no JSON 

Thanks !

**Comment by Biboba** [comment]
Just forgot this bug has been observed on ArcGIS Enterprise 10.8.1 ?
If I do get any reply, does that mean that you consider it as not a bug ? Is it more on ArcGIS Enterprise side ? Should I open an official ESRI ticket because ArcGIS Enterprise should return at least a JSON ? Were you not able to reproduce the issue ?
Thanks !

**Comment by walesdata** [comment]
I'm getting this same error when calling GIS() on any URL of our ArcGIS Enterprise install.

print("ArcGIS Enterprise login proof of concept")
gis = GIS("https://enterprise.landintelligence.net", username, password)
print("Logged in as " + str(gis.properties.user.username))


Produces:

ArcGIS Enterprise login proof of concept

---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
<ipython-input-6-75c3bbee293d> in <module>
      1 print("ArcGIS Enterprise login proof of concept")
----> 2 gis = GIS("https://enterprise.landintelligence.net", "johnny", "$weetGrass18")
      3 print("Logged in as " + str(gis.properties.user.username))

/usr/local/lib/python3.7/dist-packages/arcgis/gis/__init__.py in __init__(self, url, username, password, key_file, cert_file, verify_cert, set_active, client_id, profile, **kwargs)
    337                                    "argument when connecting to the GIS.")
    338             else:
--> 339                 raise e
    340         try:
    341             if url.lower().find("arcgis.com") > -1 and \

/usr/local/lib/python3.7/dist-packages/arcgis/gis/__init__.py in __init__(self, url, username, password, key_file, cert_file, verify_cert, set_active, client_id, profile, **kwargs)
    321                                            expiration=self._expiration,
    322                                            referer=self._referer,
--> 323                                            custom_auth=custom_auth)
    324             if self._is_hosted_nb_home:
    325                 # For GIS("home") objects, force no referer passed in

/usr/local/lib/python3.7/dist-packages/arcgis/gis/_impl/_portalpy.py in __init__(self, url, username, password, key_file, cert_file, expiration, referer, proxy_host, proxy_port, connection, workdir, tokenurl, verify_cert, client_id, custom_auth, token)
    165                                       client_id=client_id,
    166                                       custom_auth=custom_auth,
--> 167                                       token=token)
    168         #self.get_version(True)
    169         self.get_properties(True)

/usr/local/lib/python3.7/dist-packages/arcgis/gis/_impl/_con/_connection.py in __init__(self, baseurl, username, password, **kwargs)
    187 
    188         else:
--> 189             self._product = self._check_product()
    190         self._baseurl = self._validate_url(self._baseurl)
    191         self.baseurl = self._baseurl

/usr/local/lib/python3.7/dist-packages/arcgis/gis/_impl/_con/_connection.py in _check_product(self)
   1411                 res = self.get(baseurl + 'info', params={'f' : 'json'}, add_token=False)
   1412             else:
-> 1413                 res = self.get(baseurl + '/info', params={'f' : 'json'}, add_token=False)
   1414             if self._token_url is None and \
   1415                res is not None and \

/usr/local/lib/python3.7/dist-packages/arcgis/gis/_impl/_con/_connection.py in get(self, path, params, **kwargs)
    423                 'An unknown error occurred: %s' % traceback.format_exc())
    424 
--> 425         return self._handle_response(resp, file_name, out_path, try_json, force_bytes=kwargs.pop('force_bytes', False))
    426     #----------------------------------------------------------------------
    427     def _handle_response(self, resp, file_name, out_path, try_json, force_bytes=False):

/usr/local/lib/python3.7/dist-packages/arcgis/gis/_impl/_con/_connection.py in _handle_response(self, resp, file_name, out_path, try_json, force_bytes)
    495                     raise Exception(resp['error'])
    496             else:
--> 497                 data = resp.json()
    498             #if 'error' in data:
    499                 #raise Exception(data['error'])

/usr/local/lib/python3.7/dist-packages/requests/models.py in json(self, **kwargs)
    824                     # used.
    825                     pass
--> 826         return complexjson.loads(self.text, **kwargs)
    827 
    828     @property

/usr/lib/python3.7/json/__init__.py in loads(s, encoding, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    346             parse_int is None and parse_float is None and
    347             parse_constant is None and object_pairs_hook is None and not kw):
--> 348         return _default_decoder.decode(s)
    349     if cls is None:
    350         cls = JSONDecoder

/usr/lib/python3.7/json/decoder.py in decode(self, s, _w)
    335 
    336         """
--> 337         obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338         end = _w(s, end).end()
    339         if end != len(s):

/usr/lib/python3.7/json/decoder.py in raw_decode(self, s, idx)
    353             obj, end = self.scan_once(s, idx)
    354         except StopIteration as err:
--> 355             raise JSONDecodeError("Expecting value", s, err.value) from None
    356         return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)

**Comment by priyankatuteja** [comment]
@Biboba Apologies for the delay in response! I have been able to reproduce this error too. Thanks for reporting it. 

@achapkowski I get the same error with any of the following commands

```
ent_gis.admin.idp.properties
ent_gis.admin.idp.configuration
```

**Comment by nanaeaubry** [comment]
This will be fixed in 2.1.0 release. Thanks!
